tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/UIWorkingSetWizardsAuto.java

```diff
@@ -99,11 +100,42 @@ private void deleteResources() {
 
 		} catch (CoreException e) {
 			TestPlugin.getDefault().getLog().log(e.getStatus());
-			throw new AssertionError(e);
+			throw createAssertionError(e);
 		}
 
 	}
 
+	private AssertionError createAssertionError(CoreException originalException) {
+		Throwable cause = originalException.getCause();
+		if (cause == null) {
+			IStatus mostSevere = findMostSevere(originalException.getStatus());
+			cause = mostSevere.getException();
+		}
+
+		return new AssertionError(originalException.getMessage(), cause);
+	}
+
+	private IStatus findMostSevere(IStatus status) {
+		if (!status.isMultiStatus()) {
+			return status;
+		}
+
+		IStatus mostSevere = null;
+
+		for (IStatus c : status.getChildren()) {
```

Comment:
Maybe you can call this something like `childStatus` rather than `c`to make it easier to read the following line.

Reply:
Done in 1d2565a